src/openforms/formio/datastructures.py

```diff
@@ -42,6 +42,20 @@ def __contains__(self, key: str) -> bool:
     def __getitem__(self, key: str) -> Component:
         return self.component_map[key]
 
+    def __add__(self, configuration: JSONObject):
+        new_components = configuration["components"]
+        self._configuration["components"] += new_components
+
+        if self._cached_component_map is not None:
+            self._cached_component_map = {
+                **self._cached_component_map,
+                **{
+                    component["key"]: component
+                    for component in iter_components(configuration, recursive=True)
+                },
+            }
```

Comment:
if you do `def __add__(self, other: "FormioConfigurationWrapper"):`, you could leverage the properties of that instance:

```py
self.component_map.update(other.component_map)
```

otherwise, a simple for loop would be more readable:

```py
for component in iter_components(configuration, recursive=True):
    self._cached_component_map[component["key"]] = component
```